beacon-chain/skip-sync-network.md

```diff
@@ -0,0 +1,55 @@
+## Status
+>  This specification is a work-in-progress and should be considered preliminary.
+
+
+## DHT Overview
+A client uses `LightClientSkipSyncUpdate` to skip sync from a known header to a recent header. A client with a trusted but outdated header cannot use the messages in the gossip channel `bc-light-client-update` to update. The client's sync-committee in the stored snapshot is too old and not connected to any update messages. The client look for the appropriate `LightClientSkipSyncUpdate` to skip sync its header.
+
+The subprotocol id is: `0x501A`.
+
+
+## Wire Protocol
+The wire protocol](../portal-wire-protocol.md) specifies the generic wire formats of each of message type: PING, PONG, FINDNODES, FOUNDNODES, OFFER, ACCEPT, FINDCONTENT, and CONTENT. Defining the following functions allow the client to construct the wire formats of the message type `LightClientSkipSyncUpdate`.
+
+1. `content_key`
+1. `content_id` 
+1. `payload`
+
+
+```python
+class LightClientSkipSyncUpdate(Container):
+    # Update beacon block header
+    header: BeaconBlockHeader
+    # Current sync committee corresponding to the header
+    current_sync_committee: SyncCommittee
+    current_sync_committee_branch: Vector[Bytes32, floorlog2(NEXT_SYNC_COMMITTEE_INDEX)]
+    # Next sync committee corresponding to the header
+    next_sync_committee: SyncCommittee
+    next_sync_committee_branch: Vector[Bytes32, floorlog2(NEXT_SYNC_COMMITTEE_INDEX)]
+    # Sync committee aggregate signature
+    sync_committee_bits: Bitvector[SYNC_COMMITTEE_SIZE]
+    sync_committee_signature: BLSSignature
+    # Fork version for the aggregate signature
+    fork_version: Version
+
+class SkipSyncUpdateKey(Container):
+    epoch: Epoch
+    current_sync_committee: SyncCommittee
+    next_sync_committee: SyncCommittee
+    fork_version: Version
```

Comment:
Why do you need to send the `current_sync_committee` and `next_sync_committee` as part of the request? The only data the update provider needs is
- What's your current sync period you have a trusted `SyncCommittee` for
- Up to what sync period do you want to go

So something like
```python
class SkipSyncUpdateKey(Container):
    from_epoch: Epoch
    to_epoch: Epoch
```

If we want to batch. Otherwise if you want to request a single update you just need to send an epoch
```python
class SkipSyncUpdateKey(Container):
    epoch: Epoch
```

Why send the `fork_version`?

Reply:
Just `from_epoch` and `to_epoch` do not uniquely identify the object. There could be other `LightClientSkipSyncUpdate` instances; for example, one could make a LightClientSkipSyncUpdate with a fake `current_sync_committee`.

I made a mistake though. We do not need `current_sync_committee`.
```
    epoch: Epoch
    next_sync_committee: SyncCommittee
    fork_version: Version
```

Reply:
The point of the `LightClientSkipSyncUpdate` is to learn the next sync committee. How can you send that as a request parameter without creating a circular dependency?

Reply:
I still don't understand why you have to send the full `SyncCommittee` obj to identify data that is perfectly identifiable with just `epoch`. There will always exist a single `sync_committeee` for each `epoch`.

Reply:
`epoch` is uniquely identifying only if everyone has the same canonical chain. There could be multiple update objects correspond to a particular epoch if there are dishonest information on the network to try to guide users down a bad updating path.

Reply:
As discussed offline, identifying a sync update with a DHT style network will be tricky because the data is fundamentally not unique. For efficiency the network should only mantain a single update per sync period, since more is useless. The question is how to choose this one update. The best update for a consumer is:
- Finalized update
- With the most participation (highest bit count is aggregate)

Another tie breaker condition would be the most recent. But for historical sync it doesn't really matter at all. If all bridge nodes follow this conditions strictly and see all blocks in the network they should come to consensus on what's this best update per period. This may not be the case for on-going periods

Reply:
@jinfwhuang Mentioned the need for relayers to validate updates (if I understood correctly). I'm afraid that's not possible without keeping a record of at least one historical update per period, so be able to verify the signature in that update.

Reply:
We had further offline discussions on this item.

A few comments:
1. We should write out the rules for picking the best update per sync period. @dapplion has already written some down loosely in the previous comment.
2. DHT nodes needs to be able to validate each message's internal consistency. That is, the `Update` message contains valid BLS signatures agreeing with the current sync-committee. That would require each message key to contain at least a fingerprint of the current sync committee. This would prevent someone from being able to pollute the DHT with invalid messages.

I will update the PR to address these issues.